elm-src/NixJSON.elm

```diff
@@ -0,0 +1,150 @@
+module NixJSON exposing (Gson, gnull, nixFromGson, gsonDecoder)
+
+import Dict exposing (Dict, empty, foldr, filter)
+import Json.Decode as Decode
+
+type Gson
+    = GString String
+    | GBool Bool
+    | GInt Int
+    | GFloat Float
+    | GList (List Gson)
+    | GDict (Dict String Gson)
+    | GNull
+
+gnull : Gson
+gnull = GNull
+
+nixFromGson : Gson -> String
+nixFromGson gson =
+    case gson of
+        GNull ->
+            "null"
+
+        GString x ->
+            nix_string x
+
+        GBool True ->
+            "true"
+
+        GBool False ->
+            "false"
+
+        GInt x ->
+            toString x
+
+        GFloat x ->
+            toString x
+
+        GList list ->
+            nix_list list
+
+        GDict dict ->
+            if ((Dict.member "_type" dict) && ((Dict.get "_type" dict) == Just (GString "literalExample")) && (Dict.member "text" dict)) then
+                case Dict.get "text" dict of
+                    Just (GString txt) ->
+                        txt
+
+                    otherwise ->
+                        "Literal example failed to produce text"
+            else
+                let
+                    entries =
+                        Dict.toList dict
+                in
+                    if List.length entries == 0 then
+                        "{ }"
+                    else
+                        String.concat
+                            (List.concat
+                                [ [ "{\n" ]
+                                , indent
+                                    (List.concat
+                                        (List.map nix_enc_dict entries)
+                                    )
+                                , [ "}" ]
+                                ]
+                            )
+
+
+nix_string : String -> String
+nix_string x =
+    if String.contains "\n" x then
+        String.concat
+            (List.concat
+                [ [ "''\n" ]
+                , [ x ]
+                    |> segments_to_lines
+                    |> indent
+                , [ "''" ]
+                ]
+            )
+    else
+        String.concat [ "\"", x, "\"" ]
```

Comment:
I really do think `"\"" ++ x ++ "\""` reads easier, but perhaps that's just personal preference